### PR TITLE
Add key features

### PR DESCRIPTION
**Pull request by sshine**
This uses the `features-functional`, `features-lazy`, and `features-strongly-typed` icons present in the example text:

https://github.com/exercism/docs/blob/main/anatomy/tracks/config-json.md#key-features

The three others, `features-type-inference`, `features-type-classes`, and ~`features-product-types`~ `features-algebraic-data-types` are proposals to have icons made.

Note that they don't exactly correspond, but close enough:

- "Purely functional" becomes `features-functional`, not exact but close.
- "Non-strict" becomes `features-lazy`, obligatory [non-strict vs. lazy](https://wiki.haskell.org/Lazy_vs._non-strict) remark
- "Strong, static typic" becomes `features-strongly-typed`, quite close
- `features-type-inference` was made up by me
- `features-type-classes` was made up by me (could have been `features-typed-overloading`, but this is more precise)
- ~`features-product-types`~ was made up by me (~could have been~ *is* `features-algebraic-data-types`)

Note that:
- These were proposed in #938.
- CI is currently broken, so perhaps wait with merging this until CI has been fixed.
- I have not assessed how much CI / configlet v3 checks the validity of this.
- When merging, this should be squash merged.

@ErikSchierboom: How does this process work?

### CODE REVIEW

File: config.json
Comment (ErikSchierboom) [inline config.json:1111]: This slightly exceeds the maximum length (100).

**Review by ErikSchierboom — COMMENTED** [review]
> CI is currently broken, so perhaps wait with merging this until CI has been fixed.

This is up to you, the track maintainers. We're fine with having tracks in a broken state, as it doesn't influence exercism.io, but it does influence the syncing to exercism.lol. 

> I have not assessed how much CI / configlet v3 checks the validity of this.

It doesn't validate any of it, but I hope we have something implemented validating this soon.

Regarding the icons: once many tracks have defined key features, we'll try and figure out a set of icons that make sense.

**Comment by sshine** [comment]
> Regarding the icons: once many tracks have defined key features, we'll try and figure out a set of icons that make sense.

I want to know if I can merge key features that refer to icons that presumably don't yet exist.

**Review by ErikSchierboom — APPROVED** [review]
LGTM!

**Comment by ErikSchierboom** [comment]
> I want to know if I can merge key features that refer to icons that presumably don't yet exist.

Yes, you can.

**Comment by hsenag** [comment]
How come this didn't get merged? (just diggint around to see if there's anything I can do to quickly improve things)

**Comment by ErikSchierboom** [comment]
@hsenag @sshine Closed the PR. We'd be happy for you to open a new PR!

**Comment by hsenag** [comment]
@ErikSchierboom yeah, was just wondering why as it sounded like it was all ok! If the change still seems reasonable I shall revive it.

**Comment by hraftery** [comment]
Was thinking the same thing @hsenag . I say go for it. Can always be improved in the future since it's easier to edit than create.

Note however:

1. The icon names look wrong. From what I can see [elsewhere](https://github.com/exercism/csharp/blob/main/config.json) they shouldn't have the `features-` prefix.
2. The `features-type-inference`, `features-type-classes` and `features-product-types` icons don't exist. Given the guidance in the [spec](https://github.com/exercism/docs/blob/main/building/tracks/config-json.md) a suitable existing icon can probably be picked. I created a gallery of the icons [here](https://github.com/hraftery/website-icons/tree/main/key-features).
3. The features do indeed sound like features, but it seems the intention is more about listing the key differentiators or reasons for interest, rather than . In other words, the wording should probably be universally appealing, rather than specific to say, functional languages. Again, "it's easier to edit than create" and these are probably a fine start, and labouring over generalisations can probably be left for another time.

**Comment by hsenag** [comment]
@hraftery thanks! Despite my stated intention I got a bit stalled on the same questions you've mentioned, and I agree it makes sense to just go ahead and do something to improve later. If you were thinking about picking this up yourself please don't feel blocked by me, though I may well come back to this soon if you don't.

**Comment by hraftery** [comment]
@hsenag I managed to find a few minutes to pick this up. I've submitted a PR for review.

**Comment by petertseng** [comment]
Note that if you want the PR to be reviewed by the maintainers of this track (instead of yourself), and to be visible on the site when it gets merged (instead of only visible to yourself), you'll have to submit it to this repo, instead of your own.

**Comment by hraftery** [comment]
> Note [...] you'll have to submit it to this repo, instead of your own.

Ha! Right you are. Every time I think I've finally aligned with the Github workflow, I get another humbling wake up call. Thanks, I've tried again!

**Pull request closed without merging**
